rclcpp_child_sample/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.8)
+project(rclcpp_child_sample)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+  ament_cmake
+  rclcpp
+  rclcpp_sample
+)
+foreach(dep IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
+  find_package(${dep} REQUIRED)
+endforeach()
+
+set(app_derived app_derived)
+add_executable(${app_derived} src/exec.cpp)
+target_link_libraries(${app_derived} rclcpp_sample::rclcpp_sample)
```

Comment:
Finally `colcon build` passed. Catch was:
- In the upstream lib, export the library following [the ament tutorial (docs.ros.org)](https://docs.ros.org/en/galactic/How-To-Guides/Ament-CMake-Documentation.html#adding-dependencies).
- In downstream, call `target_link_libraries` (as the line above).
   - I do NOT yet understand why the linked target is duplicated as `rclcpp_sample::rclcpp_sample`. I only found a clue of this when skimming through the generated `rclcpp_sampleTargetsExport.cmake` in the install directory.

Reply:
Asked an assistance [robotics.stackexchange.com#24837](https://robotics.stackexchange.com/questions/24837/ament-target-dependencies-not-linking-an-upstream-ament-package) to better understand so that I can avoid this ugly macro in the future.